### PR TITLE
docs(samples): updated samples code to use async await

### PR DESCRIPTION
**Pull request by praveenqlogic**
Fixes googleapis/google-cloud-node#2869 (it's a good idea to open an issue first for discussion)
Fixes https://github.com/googleapis/nodejs-dataproc

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/quickstart.js [hunk 1]
```diff
@@ -14,60 +14,61 @@
...
+    // const rawResponse = responses[2];
+    for (let i = 0; i < resources.length; i += 1) {
+      console.log(resources[i]);
+    }
+    if (nextRequest) {
+      // Fetch the next page.
+      return client.listClusters(nextRequest, options).then(callback);
+    }
+  };
+  client.listClusters(request, options).then(callback);
```
Comment (ajaaym) [inline samples/quickstart.js]: why no await here?

File: samples/quickstart.js [hunk 2]
```diff
@@ -14,60 +14,61 @@
...
+    // The next request if the response shows that there are more responses.
+    const nextRequest = responses[1];
+    // The actual response object, if necessary.
+    // const rawResponse = responses[2];
+    for (let i = 0; i < resources.length; i += 1) {
+      console.log(resources[i]);
+    }
+    if (nextRequest) {
+      // Fetch the next page.
+      return client.listClusters(nextRequest, options).then(callback);
```
Comment (ajaaym) [inline samples/quickstart.js]: why no await here?

**Review by ajaaym — CHANGES_REQUESTED** [review]

File: samples/quickstart.js (hunk 1 above)
Comment (praveenqlogic) [reply samples/quickstart.js]: resolved!

File: samples/quickstart.js (hunk 2 above)
Comment (praveenqlogic) [reply samples/quickstart.js]: resolved

**Review by ajaaym — APPROVED** [review]

**Pull request merged**